src/thing_creature.c

```diff
@@ -3833,34 +3833,51 @@ TbBool creature_increase_level(struct Thing *thing)
   return false;
 }
 
-TbBool creature_increase_multiple_levels(struct Thing *thing, int count)
+TbBool creature_change_multiple_levels(struct Thing *thing, int count)
 {
     struct CreatureControl* cctrl = creature_control_get_from_thing(thing);
     if (creature_control_invalid(cctrl))
     {
         ERRORLOG("Invalid creature control; no action");
         return false;
-  }
-  struct Dungeon* dungeon = get_dungeon(thing->owner);
-  int k = 0;
-  for (int i = 0; i < count; i++)
-  {
-    if (dungeon->creature_max_level[thing->model] > cctrl->explevel)
+    }
+    struct Dungeon* dungeon = get_dungeon(thing->owner);
+    int k = 0;
+    if (count > 0)
     {
-        struct CreatureStats* crstat = creature_stats_get_from_thing(thing);
-        if ((cctrl->explevel < CREATURE_MAX_LEVEL - 1) || (crstat->grow_up != 0))
+        for (int i = 0; i < count; i++)
         {
-            cctrl->spell_flags |= CSAfF_ExpLevelUp;
-            update_creature_levels(thing);
-            k++;
-      }
+            if (dungeon->creature_max_level[thing->model] > cctrl->explevel)
+            {
+                struct CreatureStats* crstat = creature_stats_get_from_thing(thing);
+                if ((cctrl->explevel < CREATURE_MAX_LEVEL - 1) || (crstat->grow_up != 0))
+                {
+                    cctrl->spell_flags |= CSAfF_ExpLevelUp;
+                    update_creature_levels(thing);
+                    k++;
+                }
+            }
+        }
+        if (k != 0)
+        {
+            return true;
+        }
+        return false;
+    }
+    else
+    {
+        remove_creature_score_from_owner(thing);
+        if (cctrl->explevel < abs(count))
+        {
+            set_creature_level(thing, 0);
+        }
+        else
+        {
+            set_creature_level(thing, cctrl->explevel + count);
+        }
+        thing->health = cctrl->max_health;
```

Comment:
This resets the creature's health to full, right? Did this happen before?

Reply:
> This resets the creature's health to full, right? Did this happen before?

I see it's not needed, the set_creature_level already handles the health properly. I removed this line.